lib/HALInterfaces/src/IFileHandler.h

```diff
@@ -69,11 +69,20 @@ class IFileReader
      */
     virtual size_t readFile(const String& fileName, char* outBuffer, const uint32_t maxBufferSize) = 0;
 
+    /**
+     * Write a file to the filesystem.
+     * @param[in] fileName Name of the file to write. Name must be an absolute path.
+     * @param[in] buffer Buffer to write file from.
+     * @param[in] bufferSize Number of bytes in the buffer.
+     * @returns number of bytes written.

```

Comment:
Please be aware of empty lines between description, parameters and return value.
Please start e.g. "number of bytes ..." in uppercase.

This is comment is valid for all other places as well.